### ISSUE TITLE
Gnome 45

### ISSUE THREAD

**Issue by mariuszste** [mmai/Current_screen_only_on_window_switcher#19]
Please add support for gnome 45

**Comment by mmai** [comment]
fixed by #20

**Comment by sedax90** [comment]
any news?

**Comment by DameerGamlet** [comment]
any news?

**Comment by bramgn** [comment]
When adding to Gnome 45 and restarting Gnome, I see this message in the Extensions app: "class heritage AltTab.AppSwitcher is not an object or null".

**Comment by sedax90** [comment]
I don't understand if the problem is related to the fact that there are no commits on gnome-console to be able to finish the gnome 45 patch or if something simply needs to be updated, in the latter case I could finish it myself if @mmai  explains to me what the problem is

**Comment by mmai** [comment]
Hello
Unfortunately the gnome functions used in this extension are not publicly available anymore in gnome 45. 
It would require to override much more gnome core code to make this extension work on gnome 45, which will render the maintenance more complex as well. 

And there is now  https://extensions.gnome.org/extension/4412/advanced-alttab-window-switcher/  which do the same, and much more.

**Comment by dumbasPL** [comment]
> And there is now https://extensions.gnome.org/extension/4412/advanced-alttab-window-switcher/ which do the same, and much more.

Might be a good idea to put this in a more noticeable place if you have decided to stop maintaining this extension.

I loved the simplicity of this extension, in AAWS you have to disable like 95% of the features to get the same effect, but hey, at least it can be done now.

**Comment by mmai** [comment]
> Might be a good idea to put this in a more noticeable place if you have decided to stop maintaining this extension

It's at the top of the Readme as well.

**Comment by bramgn** [comment]
> I loved the simplicity of this extension, in AAWS you have to disable like 95% of the features to get the same effect, but hey, at least it can be done now.

Agreed, AAWS does way too much for my personal liking to solve a simple use-case. @mmai Thanks for the status update!

**Comment by vractal** [comment]
Maybe it was the amount of options that made me miss it, but I could not find the options in the AAWS to actually have the behaviour I have with this extension (showing only windows from the same screen AND workspace).. and sadlly from some past experiences its unlikely gnome devs would be willing to expose this APIs again by request

**Comment by omentic** [comment]
Plasma 6 is coming this February, with a few changes that seem familiar to GNOME users (unified overview, new tab switcher). I'm looking forward to it. KDE seems to take care not to break its API.

edit: it has this behavior by default! nice

**Comment by mmai** [comment]
@vractal on AATWS you have to set `Default filter` to `Current Monitor` on the `Window switcher` tab. The behavior is a little different than this extension : it shows the windows of the current screen if there are many windows, but if there is only one window on the current screen it shows all the windows of the current workspace, and if there is only one window on the current workspace it shows the windows of all the workspaces.

**Comment by dishcandanty** [comment]
To add on this for `AATWS` you also need to change your default alt-tab keyboard shortcut to actually get the same behavior.

**Comment by thmatosbr** [comment]
I’ve developed an extension that offers the same functionalities and is compatible with GNOME 45 and 46. You can check it out here: [Current Monitor Window/App Switcher](https://extensions.gnome.org/extension/7460/current-monitor-windowapp-switcher/).
